### PR TITLE
Update paypal command to be compatible for Django 1.9+

### PR DESCRIPTION
**Pull request by adeelkhan**
LEARNER-1245 - Subtask for LEARNER-480
Make Paypal command to be compatible with Django 1.9+

### CODE REVIEW

**Review by schenedx — APPROVED** [review]
👍 Good work

File: ecommerce/extensions/payment/management/commands/paypal_profile.py [hunk 1]
```diff
@@ -49,13 +59,12 @@ def handle(self, *args, **options):
...
 
-        if len(args) < 2:
-            raise CommandError("Required arguments `partner` and `action` are missing")
+        partner = options.get('partner')
+        action = options.get('action')
 
-        partner = args.pop(0)
-        action = args.pop(0)
+        if not partner and not action:
+            raise CommandError("Required arguments `partner` and `action` are missing")
```
Comment (clintonb) [inline ecommerce/extensions/payment/management/commands/paypal_profile.py]: Single quotes

File: ecommerce/extensions/payment/management/commands/paypal_profile.py [hunk 2]
```diff
@@ -136,26 +145,26 @@ def handle_list(self, args):  # pylint: disable=unused-argument
             pass
         self.print_json(result)
 
-    def handle_create(self, args):
+    def handle_create(self, options):
         """Wrapper for paypalrestsdk Create operation."""
-        profile_data = json.loads(self._get_argument(args, 'json string', 'create'))
+        profile_data = json.loads(self._get_argument([options.get('json')], 'json string', 'create'))
```
Comment (clintonb) [inline ecommerce/extensions/payment/management/commands/paypal_profile.py]: `_get_argument` is not needed. You're literally creating a list to accommodate an unnecessary method. Remove the method, and add proper validation on the arguments to ensure all required parameters are set (depending on the command).

**Review by clintonb — CHANGES_REQUESTED** [review]

**Comment by adeelkhan** [comment]
@clintonb Made required changes, please review.

File: ecommerce/extensions/payment/management/commands/paypal_profile.py [hunk 3]
```diff
@@ -176,23 +189,29 @@ def handle_delete(self, args):
...
-    def handle_disable(self, args):
+    def handle_disable(self, options):
         """
         Given the id of an existing web profile, find and delete any references to it
         in the database.  This reverses the effect of `handle_enable` above.
         """
-        profile_id = self._get_argument(args, 'profile_id', 'disable')
+        if not options.get('profile_id'):
+            raise CommandError("Action `{}` requires a {} to be specified.".format(options.get('action'),
+                                                                                   'profile_id'))
```
Comment (clintonb) [inline ecommerce/extensions/payment/management/commands/paypal_profile.py]: `'profile_id'` is a constant. It should not be a `kwarg`.

File: ecommerce/extensions/payment/management/commands/paypal_profile.py [hunk 4]
```diff
@@ -176,23 +189,29 @@ def handle_delete(self, args):
...
         """
         Given the id of an existing web profile, save a reference to it in the database.
 
         When PayPal checkouts are set up, we can look this profile up by name and, if
         found, specify its id in our API calls to customize the payment page accordingly.
         """
-        profile_id = self._get_argument(args, 'profile_id', 'enable')
+        if not options.get('profile_id'):
+            raise CommandError("Action `{}` requires a {} to be specified.".format(options.get('action'),
+                                                                                   'profile_id'))
```
Comment (clintonb) [inline ecommerce/extensions/payment/management/commands/paypal_profile.py]: `'profile_id'` is a constant. It should not be a `kwarg`.

File: ecommerce/extensions/payment/management/commands/paypal_profile.py [hunk 5]
```diff
@@ -164,7 +174,10 @@ def handle_delete(self, args):
         with an error, since leaving things in that state would cause the application
         to send invalid profile ids to PayPal, causing errors.
         """
-        profile_id = self._get_argument(args, 'profile_id', 'delete')
+        if not options.get('profile_id'):
+            raise CommandError("Action `{}` requires a {} to be specified.".format(options.get('action'),
+                                                                                   'profile_id'))
```
Comment (clintonb) [inline ecommerce/extensions/payment/management/commands/paypal_profile.py]: `'profile_id'` is a constant. It should not be a `kwarg`.

File: ecommerce/extensions/payment/management/commands/paypal_profile.py [hunk 6]
```diff
@@ -136,26 +134,38 @@ def handle_list(self, args):  # pylint: disable=unused-argument
...
         self.print_json(profile.to_dict())
 
-    def handle_update(self, args):
+    def handle_update(self, options):
         """Wrapper for paypalrestsdk Update operation.  This completely replaces the value of the existing profile."""
-        profile_id = self._get_argument(args, 'profile_id', 'update')
-        profile_data = json.loads(self._get_argument(args, 'json string', 'update'))
+        if not options.get('profile_id'):
+            raise CommandError("Action `{}` requires a {} to be specified.".format(options.get('action'),
+                                                                                   'profile_id'))
```
Comment (clintonb) [inline ecommerce/extensions/payment/management/commands/paypal_profile.py]: `'profile_id'` is a constant. It should not be a `kwarg`.

File: ecommerce/extensions/payment/management/commands/paypal_profile.py [hunk 7]
```diff
@@ -136,26 +134,38 @@ def handle_list(self, args):  # pylint: disable=unused-argument
...
+        if not options.get('json'):
+            raise CommandError("Action `{}` requires a {} to be specified.".format(options.get('action'),
+                                                                                   'json string'))
```
Comment (clintonb) [inline ecommerce/extensions/payment/management/commands/paypal_profile.py]: `'json string'` is a constant. It should not be a `kwarg`. It should also be, `'JSON string'`.

File: ecommerce/extensions/payment/management/commands/paypal_profile.py [hunk 8]
```diff
@@ -136,26 +134,38 @@ def handle_list(self, args):  # pylint: disable=unused-argument
...
         profile = self._do_create(profile_data)
         self.print_json(profile.to_dict())
 
-    def handle_show(self, args):
+    def handle_show(self, options):
         """Wrapper for paypalrestsdk Find operation."""
-        profile_id = self._get_argument(args, 'profile_id', 'show')
+        if not options.get('profile_id'):
+            raise CommandError("Action `{}` requires a {} to be specified.".format(options.get('action'),
+                                                                                   'profile_id'))
```
Comment (clintonb) [inline ecommerce/extensions/payment/management/commands/paypal_profile.py]: `'profile_id'` is a constant. It should not be a `kwarg`.

File: ecommerce/extensions/payment/management/commands/paypal_profile.py [hunk 9]
```diff
@@ -136,26 +134,38 @@ def handle_list(self, args):  # pylint: disable=unused-argument
             pass
         self.print_json(result)
 
-    def handle_create(self, args):
+    def handle_create(self, options):
         """Wrapper for paypalrestsdk Create operation."""
-        profile_data = json.loads(self._get_argument(args, 'json string', 'create'))
+        if not options.get('json'):
+            raise CommandError("Action `{}` requires a {} to be specified.".format(options.get('action'),
+                                                                                   'json string'))
```
Comment (clintonb) [inline ecommerce/extensions/payment/management/commands/paypal_profile.py]: `'json string'` is a constant. It should not be a `kwarg`. It should also be, `'JSON string'`.

**Review by clintonb — APPROVED** [review]
Please address the comments before merging.

**Pull request merged**